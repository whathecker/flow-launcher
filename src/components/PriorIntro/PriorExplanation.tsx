import React from "react";
import { StyleSheet, Image, Dimensions } from "react-native";
import { View, Text } from "../Themed";
import { Container, Typography } from "../../styles";

const PriorExplanation: React.FC = () => {
  return (
    <View>
      <View style={styles.explanationBlockWrapper}>
        <View style={styles.imageAreaWrapper}>
          <Image
            style={styles.image}
            source={require(`../../../assets/images/alarm-clock_23f0.png`)}
          />
        </View>
        <View style={styles.instructionAreaWrapper}>
          <Text style={styles.instructionHeader}>{`Is this task urgent?`}</Text>
          <Text style={styles.instructionSubHeader}>
            {`Answer ‘yes’ if there is any time pressure to finish the task.`}
          </Text>
        </View>
      </View>
      <View style={styles.explanationBlockWrapper}>
        <View style={styles.imageAreaWrapper}>
          <Image
            style={styles.image}
            source={require(`../../../assets/images/exclamation-mark_2757.png`)}
          />
        </View>
        <View style={styles.instructionAreaWrapper}>
          <Text
            style={styles.instructionHeader}
          >{`Is this task important?`}</Text>
          <Text style={styles.instructionSubHeader}>
            {`Think twice, and mark ‘yes’ if the task is crucial to achieving your goal.`}
          </Text>
        </View>
      </View>
    </View>
  );
};

const screen = Dimensions.get("screen");

function _getImageSize(width: number): number {
  if (width > 420) {
    return 60;
  } else {
    return 52;
  }
}

function _getHeaderTextSize(width: number): number {
  if (width > 420) {
    return 18;
  } else {
    return 16;
  }
}

function _getSubHeaderTextSize(width: number): number {
  if (width > 420) {
    return 15;
  } else {
    return 13;
  }
}

const styles = StyleSheet.create({
  wrapper: {
    flex: 1,
    flexDirection: "column",
  },
  explanationBlockWrapper: {
    flex: 5,
    ...Container.flexStart,
    width: "70%",
    marginTop: 5,
    marginBottom: 5,
  },
  imageAreaWrapper: {
    width: "30%",
  },
  image: {
    width: _getImageSize(screen.width),
    height: _getImageSize(screen.width),
  },
  instructionAreaWrapper: {
    width: "70%",
  },
  instructionHeader: {
    ...Typography.h4,
    fontSize: _getHeaderTextSize(screen.width),
    marginBottom: 6,
  },
  instructionSubHeader: {
    ...Typography.p,
    fontSize: _getSubHeaderTextSize(screen.width),
  },
});

export default PriorExplanation;
