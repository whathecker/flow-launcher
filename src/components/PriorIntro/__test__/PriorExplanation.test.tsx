import * as React from "react";
import renderer from "react-test-renderer";
import PriorExplanation from "../PriorExplanation";

describe("Test PriorExplanation component", () => {
  it("Should renders PriorExplanation correctly", () => {
    const tree = renderer.create(<PriorExplanation />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
