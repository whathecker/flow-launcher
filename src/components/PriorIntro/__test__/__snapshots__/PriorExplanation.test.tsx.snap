// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test PriorExplanation component Should renders PriorExplanation correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
      },
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#fff",
        },
        Object {
          "alignItems": "center",
          "flex": 5,
          "flexDirection": "row",
          "justifyContent": "flex-start",
          "marginBottom": 5,
          "marginTop": 5,
          "width": "70%",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#fff",
          },
          Object {
            "width": "30%",
          },
        ]
      }
    >
      <Image
        source={1}
        style={
          Object {
            "height": 60,
            "width": 60,
          }
        }
      />
    </View>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#fff",
          },
          Object {
            "width": "70%",
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#554F4F",
            },
            Object {
              "fontFamily": "Lexend-SemiBold",
              "fontSize": 18,
              "marginBottom": 6,
            },
          ]
        }
      >
        Is this task urgent?
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#554F4F",
            },
            Object {
              "fontFamily": "Lexend-Regular",
              "fontSize": 15,
            },
          ]
        }
      >
        Answer ‘yes’ if there is any time pressure to finish the task.
      </Text>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#fff",
        },
        Object {
          "alignItems": "center",
          "flex": 5,
          "flexDirection": "row",
          "justifyContent": "flex-start",
          "marginBottom": 5,
          "marginTop": 5,
          "width": "70%",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#fff",
          },
          Object {
            "width": "30%",
          },
        ]
      }
    >
      <Image
        source={1}
        style={
          Object {
            "height": 60,
            "width": 60,
          }
        }
      />
    </View>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#fff",
          },
          Object {
            "width": "70%",
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#554F4F",
            },
            Object {
              "fontFamily": "Lexend-SemiBold",
              "fontSize": 18,
              "marginBottom": 6,
            },
          ]
        }
      >
        Is this task important?
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#554F4F",
            },
            Object {
              "fontFamily": "Lexend-Regular",
              "fontSize": 15,
            },
          ]
        }
      >
        Think twice, and mark ‘yes’ if the task is crucial to achieving your goal.
      </Text>
    </View>
  </View>
</View>
`;
